src/include/migraphx/op/if_op.hpp

```diff
@@ -70,11 +70,11 @@ struct if_op
         module_ref mod = cond ? mods[0] : mods[1];
         std::unordered_map<std::string, argument> params;
 
-        std::set<std::string> pnames;
+        std::vector<std::string> pnames;
         for(const_module_ref smod : mods)
         {
             auto names = smod->get_parameter_names();
-            pnames.insert(names.begin(), names.end());
+            pnames.insert(pnames.end(), names.begin(), names.end());
```

Comment:
This will duplicate the arguments when both submodules have the same parameters. 

In the most likely case, the then module will have parameter 'x', and else module will also have parameter 'x'. So its expected that the user can just pass `if(cond, x)` not `if(cond, x, x)`.

Reply:
So `pnames` "vector" should be de-duplicated